### PR TITLE
build-script: add a flag to skip building/installing swift-syntax's swift side

### PR DESCRIPTION
**Pull request by nkcsgexi**

### CODE REVIEW

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

File: utils/build-script-impl [hunk 1]
```diff
@@ -83,6 +83,9 @@ KNOWN_SETTINGS=(
     xctest-build-type           "Debug"          "the build variant for xctest"
     swiftpm-build-type          "Debug"          "the build variant for swiftpm"
     swiftsyntax-build-type      "Debug"          "the build variant for swiftSyntax"
+    # When this flag is set, the build-script will only build/install the libParser library.
+    # This is a walk-around of having a separate build product for swift-syntax parser
```
Comment (akyrtzi) [inline utils/build-script-impl]: typo `walk-around` -> `workaround`. Also good to add `temporary workaround`

Comment (nkcsgexi) [reply utils/build-script-impl]: Done

**Comment by nkcsgexi** [comment]
@swift-ci please smoke test

**Review by akyrtzi — APPROVED** [review]

**Pull request merged**